workspaces/ui-v2/src/pages/docs/DocumentationRootPage/EndpointRow.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { FC, useState } from 'react';
+import { useHistory, useRouteMatch } from 'react-router-dom';
+import {
+  IconButton,
+  makeStyles,
+  ListItem,
+  Tooltip,
+  darken,
+} from '@material-ui/core';
+import { Delete as DeleteIcon, Undo as UndoIcon } from '@material-ui/icons';
+
+import { RemovedRedBackground } from '<src>/constants/theme';
+import { useFeatureFlags } from '<src>/contexts/config/AppConfiguration';
+import { EndpointName } from '<src>/components';
+import { IEndpoint } from '<src>/types';
+import { getEndpointId } from '<src>/utils';
+
+import { useContributionEditing } from '../contexts/Contributions';
+import {
+  DeleteEndpointConfirmationModal,
+  EndpointNameMiniContribution,
+} from '../components';
+
+type EndpointRowProps = {
+  endpoint: IEndpoint;
+};
+
+export const EndpointRow: FC<EndpointRowProps> = ({ endpoint }) => {
+  const history = useHistory();
+  const match = useRouteMatch();
+  const classes = useStyles();
+  const endpointId = getEndpointId({
+    method: endpoint.method,
+    pathId: endpoint.pathId,
+  });
+  const { isEditing } = useContributionEditing();
+  const { showDeleteEndpointUi } = useFeatureFlags();
+  const [deleteModalOpen, setDeleteModalOpen] = useState(false);
+
+  // TODO redux-delete-implement replace this with redux selector query
```

Comment:
Will address this in a follow up PR (which will involve moving the contributions editing flow into redux and then attaching delete state into that)